pytorch_lightning/loggers/neptune.py

```diff
@@ -46,235 +98,437 @@ class NeptuneLogger(LightningLoggerBase):
-            experiment_name="default",  # Optional,
-            params={"max_epochs": 10},  # Optional,
-            tags=["pytorch-lightning", "mlp"],  # Optional,
-        )
-        trainer = Trainer(max_epochs=10, logger=neptune_logger)
-
-    **OFFLINE MODE**
-
-    .. testcode::
-
-        from pytorch_lightning.loggers import NeptuneLogger
-
-        # arguments made to NeptuneLogger are passed on to the neptune.experiments.Experiment class
-        neptune_logger = NeptuneLogger(
-            offline_mode=True,
-            project_name="USER_NAME/PROJECT_NAME",
-            experiment_name="default",  # Optional,
-            params={"max_epochs": 10},  # Optional,
-            tags=["pytorch-lightning", "mlp"],  # Optional,
+            project="common/new-pytorch-lightning-integration",
+            name="lightning-run",  # Optional
         )
         trainer = Trainer(max_epochs=10, logger=neptune_logger)
 
-    Use the logger anywhere in you :class:`~pytorch_lightning.core.lightning.LightningModule` as follows:
+    Use the logger anywhere in your :class:`~pytorch_lightning.core.lightning.LightningModule` as follows:
 
     .. code-block:: python
 
+        from neptune.new.types import File
+
         class LitModel(LightningModule):
             def training_step(self, batch, batch_idx):
                 # log metrics
-                self.logger.experiment.log_metric("acc_train", ...)
+                acc = ...
+                self.logger.experiment["train/acc"].log(acc)
+
                 # log images
-                self.logger.experiment.log_image("worse_predictions", ...)
-                # log model checkpoint
-                self.logger.experiment.log_artifact("model_checkpoint.pt", ...)
-                self.logger.experiment.whatever_neptune_supports(...)
+                img = ...
+                self.logger.experiment["train/misclassified_images"].log(File.as_image(img))
 
             def any_lightning_module_function_or_hook(self):
-                self.logger.experiment.log_metric("acc_train", ...)
-                self.logger.experiment.log_image("worse_predictions", ...)
-                self.logger.experiment.log_artifact("model_checkpoint.pt", ...)
-                self.logger.experiment.whatever_neptune_supports(...)
+                # log model checkpoint
+                ...
+                self.logger.experiment["checkpoints/epoch37"].upload("epoch=37.ckpt")
 
-    If you want to log objects after the training is finished use ``close_after_fit=False``:
+                # generic recipe
+                metadata = ...
+                self.logger.experiment["your/metadata/structure"].log(metadata)
 
-    .. code-block:: python
+    Check `Neptune docs <https://docs.neptune.ai/user-guides/logging-and-managing-runs-results/logging-runs-data>`_
+    for more info about how to log various types metadata (scores, files, images, interactive visuals, CSVs, etc.).
```

Comment:
Would be neat for Netpune to natively add support for this as `log_image, log_csv, log_...`. Maybe another PR.

Reply:
Ok, since you don't have clear interface in [Standardize API for logging proposal](https://github.com/PyTorchLightning/pytorch-lightning/issues/6720), I'll support only method which were available in previous version of `NeptuneClient`.

Once you decide what set of `log_*` functions you want to keep, we'll implement it :)

Reply:
Sounds great ! Thanks.